### PR TITLE
[feat] refactor CLI implementation and change commands API

### LINKED ISSUE #2: General CLI improvements
I was looking through the source code and was thinking that it might be a good idea to use some lightweight framework for building CLI. It should make it easier to maintain current and add new commands in the future. It would also simplify things like help command, arguments parsing, validation etc.

Please, let me know your thoughts about it.

### PR DESCRIPTION
**Pull request by mrkldshv**
* Removes colon prefix from all commands (`lazy :run test` -> `lazy run test`)
* Removes option to forwards args without `--` from root command (`lazy test --runInBand` -> `lazy test -- --runInBand`)
* Adds [cac](https://github.com/cacjs/cac) library to build CLI

Closes #2.

### CODE REVIEW

**Comment by mrkldshv** [comment]
I decided to go with cac library (it's used by vite projects). It seemed like a good choice for our case because it's lightweight and has simple API, albeit not as feature-rich as other libraries like yargs and commander.

I'm open to any alternatives, especially if we're planning have lots of other commands and options.

**Comment by ds300** [comment]
Thanks Mark! Had a quick look, and seems great. I don't have time for a proper review right now, will take a deeper look on Sunday.

**Comment by mrkldshv** [comment]
Sure, will be waiting for any comments!

File: src/cli.js [hunk 1]
```diff
@@ -1,60 +1,49 @@
...
+  process.exit(0)
+})
+
+cli
+  .command('inherit', 'run command from configuration file specified by script name')
+  .action(async () => {
+    await inherit()
+  })
+
+cli.help()
```
[the remark this answers is not in the document]
Comment (ds300) [reply src/cli.js:45]: I pushed a commit fixing this. The problem was that the interactive logger registered a setInterval and didn't clean it up, so I make it create the setInterval lazily and clean it up at the end.

**Review by ds300 — APPROVED** [review]
This seems to work well! I added a catch-all to print the help message when the user fails to supply a command. I think we can also delete the help.js file now. 

I messed up the authorship metadata on the commits while rebasing. If you wanna redo the commits as yourself and force push, lemme know.

Otherwise I'll merge this tomorrow 👍🏼 Thanks again 🙇🏼

File: src/cli.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrkldshv) [reply src/cli.js:45]: Got it. Thanks for the explanation!

**Pull request merged**